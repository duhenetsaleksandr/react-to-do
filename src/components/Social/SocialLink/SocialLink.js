import React from 'react';

const SocialLink = ({ link, title, children }) => (
    <a href={link} target="_blank" rel="noopener noreferrer" title={title}>
        { children }
    </a>
);

export default SocialLink;
