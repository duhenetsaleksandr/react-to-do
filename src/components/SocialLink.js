import React from 'react';

export default function SocialLink(props) {
    return (
        <a href={props.link} target="_blank" rel="noopener noreferrer" title={props.title}>
            {props.child?.()}
        </a>
    );
}
