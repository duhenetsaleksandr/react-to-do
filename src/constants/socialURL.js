export const telegramURL = 'https://t.me/duhenets';
export const instagramURL = 'https://www.instagram.com/duhenets_aleksandr/';
