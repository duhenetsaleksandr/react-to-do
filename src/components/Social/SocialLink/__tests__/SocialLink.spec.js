import React from 'react';
import SocialLink from '../SocialLink';
import { telegramURL } from 'constants/socialURL';

const props = {
    link: telegramURL,
    title: 'telegram',
    children: <div className="test-class">TestProps</div>,
};

describe('should render SocialLink component', () => {
    let component;
    beforeEach(() => {
        component = mount(<SocialLink {...props} />);
    });

    it('should contain one tag anchor', () => {
        const link = component.find('a');
        expect(link).toHaveLength(1);
    });

    it('should contain link props with correct value', () => {
        const linkProps = component.find('a').getElement().props.href;
        expect(linkProps).toBe(props.link);
    });

    it('should contain title props with correct value', () => {
        const titleProps = component.find('a').getElement().props.title;
        expect(titleProps).toBe(props.title);
    });

    it('should contain children props with correct value', () => {
        const childrenProps = component.find('a').getElement().props.children;
        expect(childrenProps).toEqual(props.children);
    });

    it('should target attr with equal _blank', () => {
        const target = component.find('a').getElement().props.target;
        expect(target).toBe('_blank');
    });

    it('should render SocialLink component snapshot with all props', () => {
        const component = shallow(<SocialLink {...props} />);
        expect(component).toMatchSnapshot();
    });
});
