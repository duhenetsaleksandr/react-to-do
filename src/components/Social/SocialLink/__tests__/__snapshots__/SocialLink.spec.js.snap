// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SocialLink component should render SocialLink component snapshot with all props 1`] = `
<a
  href="https://t.me/duhenets"
  rel="noopener noreferrer"
  target="_blank"
  title="telegram"
>
  <div
    className="test-class"
  >
    TestProps
  </div>
</a>
`;
